src/js/common/schemaform/review/ReviewPage.jsx

```diff
@@ -45,6 +53,29 @@ class ReviewPage extends React.Component {
     }
   }
 
+  setPagesViewed = (keys) => {
+    const viewedPages = keys.reduce((pages, key) => {
+      if (!pages.has(key)) {
+        // if we hit a page that we need to add, check to see if
+        // we haven't cloned the set yet; we should only do that once
+        if (pages === this.state.viewedPages) {
```

Comment:
No, I specifically want to do a reference equality check here. If we hit a page that wasn't already in the viewed pages above we clone the Set. That means that we can use a `===` check to see if there are any changes.